src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -146,6 +151,7 @@ public void onConnectionResumed(boolean sessionPresent) {
 
     CompletableFuture<Integer> publish(MqttMessage message, QualityOfService qos, boolean retain) {
         return connect().thenCompose((b) -> {
+            rateLimiter.acquire(1);
```

Comment:
Why not use `tryAcquire`? Is it desirable to be blocked here?

Reply:
I'll double check, but this shouldn't block the spooler thread due to the callback in the compose.

Reply:
I rewrote it so that it throttles in the spooler thread instead of the mqtt client. It now throttles on both TPS and bandwidth used. If there are multiple connections then it will choose to use the connection with the minimum wait time.